SARAALERT-1521: Fix api tests related to follow up reason and note

timwongj opened the pull request:
# Description
Jira Ticket: [SARAALERT-1521](https://tracker.codev.mitre.org/browse/SARAALERT-1521)

This PR includes changes to fix the api tests

## Important Changes
Please list important files (meaning substantial or integral to the PR) along with a list of the general changes that should be highlighted for reviewers.

`example_file.js`
- Example change (ex: refactored import function)

## Testing Recommendations
Please list any recommendations regarding what reviewers should test and if there is any specific guidance on how to test certain aspects of the PR. Be sure to mention edge cases that should be tried, particularly in the UI.

# Checklists

**Submitter:**
- [x] This PR describes why these changes were made.
- [x] This PR is into the correct branch.
- [x] This PR includes the correct JIRA ticket reference.
- [x] Comment added to the relevant JIRA ticket(s) with a link to this PR
- [x] Code diff has been reviewed (it **does not** contain: additional white space, not applicable code changes, debug statements, etc.)
- [x] If UI changes have been made, Chrome Dev Tools Lighthouse accessibility test has been executed to ensure no 508 issues were introduced.
- [x] Tests are included and test edge cases
- [x] Tests have been run locally and pass (remember to update Gemfile when applicable)
- [x] Test fixtures updated and documented as necessary


@tstrass :
- [x] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [x] The tests appropriately test the new code, including edge cases
- [x] You have tried to break the code
- [ ] If applicable, you have tested changes against a large database, and considered possible performance regressions
- [ ] Tested all recommendations listed in the "Testing Recommendations" section. The application behaves as expected with this PR.


@dubem7 :
- [x] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [x] The tests appropriately test the new code, including edge cases
- [x] You have tried to break the code
- [ ] If applicable, you have tested changes against a large database, and considered possible performance regressions
- [ ] Tested all recommendations listed in the "Testing Recommendations" section. The application behaves as expected with this PR.


@ngfreiter :
- [x] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [x] The tests appropriately test the new code, including edge cases
- [x] You have tried to break the code
- [x] If applicable, you have tested changes against a large database, and considered possible performance regressions
- [x] Tested all recommendations listed in the "Testing Recommendations" section. The application behaves as expected with this PR.

ngfreiter reviewed (changes requested):
This looks good, I know this wasn't you, but I noticed when I run the tests locally now, I get a deprecation warning. Since you're already fixing up some tests in this file could you just loop that in here:
![image](https://user-images.githubusercontent.com/52747882/125794518-a32be3bd-1c33-47a2-8dde-a79cee612561.png)

timwongj commented:
> This looks good, I know this wasn't you, but I noticed when I run the tests locally now, I get a deprecation warning. Since you're already fixing up some tests in this file could you just loop that in here:
> ![image](https://user-images.githubusercontent.com/52747882/125794518-a32be3bd-1c33-47a2-8dde-a79cee612561.png)

https://github.com/SaraAlert/SaraAlert/pull/1006/commits/500591004af398cf799bd315754a790ee0c312a7

tstrass commented on app/helpers/patient_details_helper.rb:
```diff
@@ -203,8 +207,6 @@ def custom_export_details_for_export
       pause_notifications: pause_notifications || false,
       expected_purge_ts: expected_purge_date_exp || '',
       monitoring_reason: monitoring_reason || '',
-      follow_up_reason: follow_up_reason || '',
-      follow_up_note: follow_up_note || '',
```
Why this removal?

timwongj replied:
this was simply moved here: https://github.com/SaraAlert/SaraAlert/pull/1006/files#diff-6a782ad659c7d687b8b004c762706d9c4c1ee24674305f8df71babe430abdb73R76

tstrass replied:
Shouldn't it be in both places? One is for verifying linelist export and the other is for custom export.

timwongj replied:
it gets combined here: https://github.com/SaraAlert/SaraAlert/pull/1006/files#diff-6a782ad659c7d687b8b004c762706d9c4c1ee24674305f8df71babe430abdb73R226

tstrass replied:
Ohh i was confused because your first link was to the linelist fields not the full history fields, but I see they are there as well 👍

tstrass commented:
Why the vaccines commit? Seems like that's unrelated/should be under a different ticket

timwongj commented:
> Why the vaccines [commit](https://github.com/SaraAlert/SaraAlert/pull/1006/commits/8d4d9dfa564c9e2e01bfa16b6a5b79a55db77807)? Seems like that's unrelated/should be under a different ticket

So I'm actually glad you accidentally merged @annmarieb1 's PR before mine because it coincidentally helped me discover some things I was missing in my 1521 PR and my vaccines import PR.

To address the deprecation warning @ngfreiter pointed out (use `assert_nil` instead of `assert_equal` for `nil` values), I updated @patient_2's follow up fields to not be `nil` in order to test those fields being populated properly as well.

Doing this broken the export tests which I discovered were incorrect because I had forgotten to add the follow up fields to the `full_history_details_for_export` method. So I added them and they were still failing and they were failing because I had also forgotten to add the vaccine fields and the order of those fields matter. So I then had to add the vaccine fields to the verifier and include the vaccine fields in the export (which is something I should have done in my vaccine import/export PR to begin with).

ngfreiter reviewed (approved):
This looks good to me (assuming we are fine with looping the vaccine handling in here).

tstrass commented:
> > Why the vaccines [commit](https://github.com/SaraAlert/SaraAlert/pull/1006/commits/8d4d9dfa564c9e2e01bfa16b6a5b79a55db77807)? Seems like that's unrelated/should be under a different ticket
> 
> So I'm actually glad you accidentally merged @annmarieb1 's PR before mine because it coincidentally helped me discover some things I was missing in my 1521 PR and my vaccines import PR.
> 
> To address the deprecation warning @ngfreiter pointed out (use `assert_nil` instead of `assert_equal` for `nil` values), I updated @patient_2's follow up fields to not be `nil` in order to test those fields being populated properly as well.
> 
> Doing this broken the export tests which I discovered were incorrect because I had forgotten to add the follow up fields to the `full_history_details_for_export` method. So I added them and they were still failing and they were failing because I had also forgotten to add the vaccine fields and the order of those fields matter. So I then had to add the vaccine fields to the verifier and include the vaccine fields in the export (which is something I should have done in my vaccine import/export PR to begin with).

Don't you see that was my plan all along merging the other PR first 🙃

Thanks for the explanation, and good catch.
